File: app/src/main/java/com/aandssoftware/aandsinventory/models/CustomerModel.java

package com.aandssoftware.aandsinventory.models;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;
import java.io.Serializable;
import java.util.HashMap;

public class CustomerModel implements Serializable {
  
  public static final String TABLE_CUSTOMER = "customer";
  public static final String CUSTOMER_COUNTER = "customerCounter";
  public static final String ORDER_BY_VALUE = "customerID";
  public static final String LOGIN_ORDER_BY_VALUE = "username";
  public static final String PASSWORD = "password";
  public static final String PERMISSION = "permission";
  
  @SerializedName("id")
  @Expose
  private String id;
  
  @SerializedName("customerID")
  @Expose
  private String customerID;
  
  @SerializedName("invoiceNumber")
  @Expose
  private String invoiceNumber;
  
  @SerializedName("customerName")
  @Expose
  private String customerName;
  
  @SerializedName("companyMail")
  @Expose
  private String companyMail;
  
  @SerializedName("customerGstNumber")
  @Expose
  private String customerGstNumber;
  
  @SerializedName("address")
  @Expose
  private String address;
  
  @SerializedName("customerNumber")
  @Expose
  private String customerNumber;
  
  @SerializedName("contactPerson")
  @Expose
  private String contactPerson;
  
  @SerializedName("contactPersonNumber")
  @Expose
  private String contactPersonNumber;
  
  @SerializedName("alternateNumber")
  @Expose
  private String alternateNumber;
  
  @SerializedName("description")
  @Expose
  private String description;
  
  @SerializedName("dateCreated")
  @Expose
  private long dateCreated;
  
  @SerializedName("invoiceCreatedDate")
  @Expose
  private long invoiceCreatedDate;
  
  @SerializedName("dueDate")
  @Expose
  private long dueDate;
  
  @SerializedName("imagePath")
  @Expose
  private String imagePath;
  
  @SerializedName("requirement")
  @Expose
  private String requirement;
  
  @SerializedName("tag")
  @Expose
  private String tag;
  
  @SerializedName("username")
  @Expose
  private String username;
  
  @SerializedName("password")
  @Expose
  private String password;
  
  @SerializedName("permission")
  @Expose
  private HashMap<String, String> permission;
  
  public CustomerModel() {
  
  }
  
  public String getId() {
    return id;
  }
  
  public CustomerModel setId(String id) {
    this.id = id;
    return this;
  }
  
  public String getCustomerName() {
    return customerName;
  }
  
  public CustomerModel setCustomerName(String customerName) {
    this.customerName = customerName;
    return this;
  }
  
  public String getCompanyMail() {
    return companyMail;
  }
  
  public CustomerModel setCompanyMail(String companyMail) {
    this.companyMail = companyMail;
    return this;
  }
  
  public String getCustomerGstNumber() {
    return customerGstNumber;
  }
  
  public CustomerModel setCustomerGstNumber(String customerGstNumber) {
    this.customerGstNumber = customerGstNumber;
    return this;
  }
  
  public String getAddress() {
    return address;
  }
  
  public CustomerModel setAddress(String address) {
    this.address = address;
    return this;
  }
  
  public String getCustomerNumber() {
    return customerNumber;
  }
  
  public CustomerModel setCustomerNumber(String customerNumber) {
    this.customerNumber = customerNumber;
    return this;
  }
  
  public String getContactPerson() {
    return contactPerson;
  }
  
  public CustomerModel setContactPerson(String contactPerson) {
    this.contactPerson = contactPerson;
    return this;
  }
  
  public String getContactPersonNumber() {
    return contactPersonNumber;
  }
  
  public CustomerModel setContactPersonNumber(String contactPersonNumber) {
    this.contactPersonNumber = contactPersonNumber;
    return this;
  }
  
  public String getAlternateNumber() {
    return alternateNumber;
  }
  
  public CustomerModel setAlternateNumber(String alternateNumber) {
    this.alternateNumber = alternateNumber;
    return this;
  }
  
  public String getDescription() {
    return description;
  }
  
  public CustomerModel setDescription(String description) {
    this.description = description;
    return this;
  }
  
  public long getDateCreated() {
    return dateCreated;
  }
  
  public CustomerModel setDateCreated(long dateCreated) {
    this.dateCreated = dateCreated;
    return this;
  }
  
  public String getImagePath() {
    return imagePath;
  }
  
  public CustomerModel setImagePath(String imagePath) {
    this.imagePath = imagePath;
    return this;
  }
  
  public String getCustomerID() {
    return customerID;
  }
  
  public CustomerModel setCustomerID(String customerID) {
    this.customerID = customerID;
    return this;
  }
  
  public String getRequirement() {
    return requirement;
  }
  
  public void setRequirement(String requirement) {
    this.requirement = requirement;
  }
  
  public String getTag() {
    return tag;
  }
  
  public void setTag(String tag) {
    this.tag = tag;
  }
  
  public String getInvoiceNumber() {
    return invoiceNumber;
  }
  
  public void setInvoiceNumber(String invoiceNumber) {
    this.invoiceNumber = invoiceNumber;
  }
  
  public long getInvoiceCreatedDate() {
    return invoiceCreatedDate;
  }
  
  public void setInvoiceCreatedDate(long invoiceCreatedDate) {
    this.invoiceCreatedDate = invoiceCreatedDate;
  }
  
  public long getDueDate() {
    return dueDate;
  }
  
  public void setDueDate(long dueDate) {
    this.dueDate = dueDate;
  }
  
  public String getUsername() {
    return username;
  }
  
  public void setUsername(String username) {
    this.username = username;
  }
  
  public String getPassword() {
    return password;
  }
  
  public void setPassword(String password) {
    this.password = password;
  }
  
  public HashMap<String, String> getPermission() {
    return permission;
  }
  
  public void setPermission(HashMap<String, String> permission) {
    this.permission = permission;
  }
}

File: app/src/main/java/com/aandssoftware/aandsinventory/models/CustomerModel.kt
package com.aandssoftware.aandsinventory.models

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.io.Serializable
import java.util.HashMap

class CustomerModel : Serializable {

    @SerializedName("id")
    @Expose
    var id: String? = null

    @SerializedName("customerID")
    @Expose
    var customerID: String? = null

    @SerializedName("invoiceNumber")
    @Expose
    var invoiceNumber: String? = null

    @SerializedName("customerName")
    @Expose
    var customerName: String? = null

    @SerializedName("companyMail")
    @Expose
    var companyMail: String? = null

    @SerializedName("customerGstNumber")
    @Expose
    var customerGstNumber: String? = null

    @SerializedName("address")
    @Expose
    var address: String? = null

    @SerializedName("customerNumber")
    @Expose
    var customerNumber: String? = null

    @SerializedName("contactPerson")
    @Expose
    var contactPerson: String? = null

    @SerializedName("contactPersonNumber")
    @Expose
    var contactPersonNumber: String? = null

    @SerializedName("alternateNumber")
    @Expose
    var alternateNumber: String? = null

    @SerializedName("description")
    @Expose
    var description: String? = null

    @SerializedName("dateCreated")
    @Expose
    var dateCreated: Long = 0

    @SerializedName("invoiceCreatedDate")
    @Expose
    var invoiceCreatedDate: Long = 0

    @SerializedName("dueDate")
    @Expose
    var dueDate: Long = 0

    @SerializedName("imagePath")
    @Expose
    var imagePath: String? = null

    @SerializedName("requirement")
    @Expose
    var requirement: String? = null

    @SerializedName("tag")
    @Expose
    var tag: String? = null

    @SerializedName("username")
    @Expose
    var username: String? = null

    @SerializedName("password")
    @Expose
    var password: String? = null

    @SerializedName("permission")
    @Expose
    var permission: HashMap<String, String>? = null

    @SerializedName("discountPercent")
    @Expose
    var discountPercent: Double = 0.0

    @SerializedName(DISCOUNTED_ITEM)
    @Expose
    var discountedItems: HashMap<String, String>? = null

    @SerializedName("blockedUser")
    @Expose
    var blockedUser: Boolean = false

    @SerializedName("blockedUserMessage")
    @Expose
    var blockedUserMessage: String? = null

    companion object {
        val TABLE_CUSTOMER = "customer"
        val CUSTOMER_COUNTER = "customerCounter"
        val ORDER_BY_VALUE = "customerID"
        val LOGIN_ORDER_BY_VALUE = "username"
        val PASSWORD = "password"
        val PERMISSION = "permission"
        const val DISCOUNTED_ITEM = "discountedItems"
    }
}
